crates/iota-light-client/src/verifier.rs

```diff
@@ -0,0 +1,422 @@
+            "Checkpoint sequence number does not match. Need to Sync."
+        );
+
+        // Get the committee from the previous checkpoint
+        let current_committee = prev_ckp
+            .end_of_epoch_data
+            .as_ref()
+            .ok_or(anyhow!(
+                "Expected all checkpoints to be end-of-epoch checkpoints"
+            ))?
+            .next_epoch_committee
+            .iter()
+            .cloned()
+            .collect();
+
+        // Make a committee object using this
+        Committee::new(prev_ckp.epoch().checked_add(1).unwrap(), current_committee)
+    } else {
+        // Since we did not find a small committee checkpoint we use the genesis
+        Genesis::load(config.genesis_blob_file_path())?
+            .committee()
+            .context("Cannot load Genesis")?
+    };
+
+    info!("Extracting effects and events for TID: {tid}");
+    extract_verified_effects_and_events(&full_check_point, &committee, tid)
+        .context("Cannot extract effects and events")
+}
+
+/// Get the verified checkpoint sequence number for an object.
+/// This function will verify that the object is in the transaction's effects,
+/// and that the transaction is in the checkpoint
+/// and that the checkpoint is signed by the committee
+/// and the committee is read from the verified checkpoint summary
+/// which is signed by the previous committee.
+pub async fn get_verified_checkpoint(
+    id: ObjectID,
+    config: &Config,
+) -> Result<CheckpointSequenceNumber> {
+    let iota_client = IotaClientBuilder::default()
+        .build(config.rpc_url.as_str())
+        .await?;
+    let read_api = iota_client.read_api();
+    let object_json = read_api
+        .get_object_with_options(id, IotaObjectDataOptions::bcs_lossless())
+        .await
+        .expect("Cannot get object");
+    let object = object_json
+        .into_object()
+        .expect("Cannot make into object data");
+    let object: Object = object.try_into().expect("Cannot reconstruct object");
+
+    // Lookup the transaction id and get the checkpoint sequence number
+    let options = IotaTransactionBlockResponseOptions::new();
+    let seq = read_api
+        .get_transaction_with_options(object.previous_transaction, options)
+        .await
+        .context("Cannot get transaction")?
+        .checkpoint
+        .ok_or(anyhow!("Transaction not found"))?;
+
+    // Need to authenticate this object
+    let (effects, _) = get_verified_effects_and_events(config, object.previous_transaction)
+        .await
+        .expect("Cannot get effects and events");
+
+    // check that this object ID, version and hash is in the effects
+    let target_object_ref = object.compute_object_reference();
+    effects
+        .all_changed_objects()
+        .iter()
+        .find(|object_ref| object_ref.0 == target_object_ref)
+        .ok_or(anyhow!("Object not found"))
+        .expect("Object not found");
```

Comment:
The ok_or is very useful with the expect :trollface:

Reply:
found another place as well, and also changed to `ok_or_else` in https://github.com/iotaledger/iota/pull/6135/commits/4055d8cb3ca31c21a9a1fbd1271974f441a9cd1f